.github/workflows/macOS_gcc10.yml

```diff
@@ -54,7 +54,7 @@ jobs:
           swig \
           yajl \
           zeromq
-        brew install --cask adoptopenjdk9
+        brew install openjdk@11
```

Comment:
To save you some time: I tried **many** combinations and only adoptopenjdk worked due to some upstream linking issues.

Reply:
Thank you! The problem is, this implies we are stuck on jdk 1.8 when using this image... Can't we upgrade to a more recent macOS version?

Reply:
Last time I checked this should be using AdoptOpenJDK packaged Java 16 on master, which is much more recent than 1.8. Where do you get this number from?

Also the macOS version is the latest supported image for GitHub Actions.

Reply:
```sh
brew install --cask adoptopenjdk9
```

Will use the Java 9 Version of AdoptOpenJDK, but `brew install --cask adoptopenjdk` will use the latest and `brew install --cask adoptopenjdk16` is fixed to Java 16. See [here](https://github.com/AdoptOpenJDK/homebrew-openjdk)

The Github Actions image also comes with AdoptOpenJDK versions for Java 8, 11, 12, 13 and 14 pre-installed. See [here](https://github.com/actions/virtual-environments/blob/main/images/macos/macos-10.15-Readme.md#java)

If we build with Java 11 (currently latest LTS) then we don't even need to install via homebrew. Setting `JAVA_HOME="$JAVA_HOME_11_X64"` should be enough.

Reply:
> If we build with Java 11 (currently latest LTS) then we don't even need to install via homebrew. Setting JAVA_HOME="$JAVA_HOME_11_X64" should be enough.

It does not work.

I tried to save you all some time as I tested this already: the other shipped macOS JDKs are linked improperly (upstream issue) and the only currently working one is the adoptopenjdk brew cask.

Reply:
> I tried to save you all some time as I tested this already: the other shipped macOS JDKs are linked improperly (upstream issue) and the only currently working one is the adoptopenjdk brew cask.

which seems to install jdk 1.8... at tleast that's the version used and detected by the cmake scripts